client/asset/btc/spv_wrapper.go

```diff
@@ -1030,36 +1032,50 @@ func (w *spvWallet) walletUnlock(pw []byte) error {
 	return w.Unlock(pw)
 }
 
-func (w *spvWallet) getBlockHeader(blockHash *chainhash.Hash) (*blockHeader, error) {
+// getBlockHeader gets the *blockHeader for the specified block hash. It also
+// returns a bool value to indicate whether this block is a part of main chain.
+// For orphaned blocks header.Confirmations is negative (typically -1).
+func (w *spvWallet) getBlockHeader(blockHash *chainhash.Hash) (header *blockHeader, mainchain bool, err error) {
 	hdr, err := w.cl.GetBlockHeader(blockHash)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 
 	tip, err := w.cl.BestBlock()
 	if err != nil {
-		return nil, fmt.Errorf("BestBlock error: %v", err)
+		return nil, false, fmt.Errorf("BestBlock error: %v", err)
 	}
 
 	blockHeight, err := w.cl.GetBlockHeight(blockHash)
 	if err != nil {
-		return nil, err
+		return nil, false, err
+	}
+
+	confirmations := int64(-1)
+	mainchain = w.blockIsMainchain(blockHash, blockHeight)
+	if mainchain {
+		confirmations = int64(confirms(blockHeight, tip.Height))
+	}
+	if tip.Height < blockHeight { // if tip is less, may be rolling back, so just mock dcrd/dcrwallet
+		confirmations = 0
 	}
 
 	return &blockHeader{
-		Hash:          hdr.BlockHash().String(),
-		Confirmations: int64(confirms(blockHeight, tip.Height)),
-		Height:        int64(blockHeight),
-		Time:          hdr.Timestamp.Unix(),
-	}, nil
+		Hash:              hdr.BlockHash().String(),
+		Confirmations:     confirmations,
+		Height:            int64(blockHeight),
+		Time:              hdr.Timestamp.Unix(),
+		PreviousBlockHash: hdr.PrevBlock.String(),
```

Comment:
Looks like without initialising `PreviousBlockHash` here we won't be able to to crawl back the chain to find common ancestor with mainchain (instead, we'd nil pointer dereference).